src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -40,9 +40,34 @@
 
 /**
  * <p>
- * Checks that a class that has only private constructors and has no descendant
- * classes is declared as final.
+ * Check that identifies classes that can be declared as final are explicitly made final.
+ * The following is the type of classes that can be identified:
  * </p>
+ * <ol>
+ *   <li>
+ *       Private classes with implicit constructor.
+ *   </li>
+ *   <li>
+ *       Classes with all the type of modifier which contains only private constructor.
+ *   </li>
+ * </ol>
+ * <p>
+ * Classes are skipped if:
+ * </p>
+ * <ol>
+ *   <li>
+ *       Class is already Final.
```

Comment:
Lower case `final` and wrap this and abstract with `{@code }`.

Reply:
Done